modules/oe_theme_inpage_navigation/js/inpage_navigation.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * @file
+ * ECL inpage navigation initializer.
+ */
+(function (ECL, Drupal) {
```

Comment:
We need to do two extra things in this function:
- In case the h2.ecl-u-type-heading-2 has no id, it should be generated and added so that it can be used by the inpage nav.
- We need to loop first in regions marked with a certain data attribute (e.g. data-inpage-navigation-provider) and mark all found h2 with the ecl-u-type-heading-2 class. Afterwards they should be found and taken in consideration by the current code and used in the in-page navigation.

Reply:
I've added autogeneration of ids for headers without ID attribute and `data-inpage-navigation-source-area` attribute for content block.